lib/matplotlib/pyplot.py

```diff
@@ -543,9 +531,26 @@ def make_active(event):
         _pylab_helpers.Gcf.set_active(figManager)
         figManager.canvas.figure.number = num
 
+        if _INSTALL_FIG_OBSERVER:
+            figManager.canvas.figure.add_callback(_auto_draw_if_interactive)
+
     return figManager.canvas.figure
 
 
+def _auto_draw_if_interactive(fig):
+    """
+    This is an internal helper function for making sure that auto-redrawing
+    works as intended in the plain python repl.
+
+    Parameters
+    ----------
+    fig : Figure
+        A figure object which is assumed to be associated with a canvas
+    """
+    if fig.stale and matplotlib.is_interactive():
```

Comment:
because if we are not interactive the user does not get updates?  This is just mimicking the existing pyplot behavior of `draw_if_interactive`.